UnityProject/Assets/Scripts/Objects/Machines/Gibber.cs

```diff
@@ -0,0 +1,159 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using HealthV2;
+using Mirror;
+using NaughtyAttributes;
+using Systems.Electricity;
+using UnityEngine;
+
+namespace Objects.Machines
+{
+	public class Gibber : MonoBehaviour, ICheckedInteractable<HandApply>, ICheckedInteractable<MouseDrop>, IExaminable, IRightClickable
+	{
+		[SerializeField] private ObjectContainer storage;
+		[SerializeField] private UniversalObjectPhysics physics;
+		[SerializeField] private Integrity machineIntegrity;
+		[SerializeField] private float timeToGib = 12f;
+		[SerializeField] private int produceMultiplier = 4;
+		[SerializeField, Tooltip("In watts.")] private int powerUseWhenActive = 1000;
+		[SerializeField] private int damagePerFrame = 20;
+		[SerializeField] private GameObject defaultProduce;
+		[SerializeField] private SpriteHandler lights;
+		[SerializeField] private SpriteDataSO lightsoff;
+		[SerializeField] private SpriteDataSO lightson;
+
+		private bool isRunning = false;
+
+		private Dictionary<GameObject, int> gibbed = new Dictionary<GameObject, int>();
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			// Only interact with this if it's anchored.
+			if (physics.isNotPushable == false) return false;
+			//Checks if the player is in reach + if they are able to interact with this type of object.
+			if (DefaultWillInteract.Default(interaction, side) == false) return false;
+			//For people attacking the object so they don't accidentally trip it.
+			return interaction.Intent != Intent.Harm;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (interaction.HandObject != null)
+			{
+				storage.StoreObject(interaction.HandObject);
+				return;
+			}
+
+			isRunning = !isRunning;
+			if (isRunning) StartCoroutine(GibbingTime());
+		}
+
+		public bool WillInteract(MouseDrop interaction, NetworkSide side)
+		{
+			// Only interact with this if it's anchored.
+			if (physics.isNotPushable == false) return false;
+			//Checks if the player is in reach + if they are able to interact with this type of object.
+			return DefaultWillInteract.Default(interaction, side);
+		}
+
+		public void ServerPerformInteraction(MouseDrop interaction)
+		{
+			storage.StoreObject(interaction.DroppedObject);
```

Comment:
just check can you escape from the gibber while it's running? or not

Reply:
You can, just right click on it and press "Eject Content"

Reply:
while the player being gibbed

Reply:
You can just turn it off while it's running and try ejecting yourself, tho the time to kill is very high on this that I don't see it mattering all that much.